lib/config_cat/client.ex

```diff
@@ -249,23 +259,48 @@ defmodule ConfigCat.Client do
     details
   end
 
-  defp cached_settings(%State{} = state) do
+  defp cached_config(%State{} = state) do
     %{cache_policy: policy, flag_overrides: flag_overrides, instance_id: instance_id} = state
-    local_settings = OverrideDataSource.overrides(flag_overrides)
+    local_config = OverrideDataSource.overrides(flag_overrides)
 
     case OverrideDataSource.behaviour(flag_overrides) do
       :local_only ->
-        {:ok, local_settings, 0}
+        {:ok, local_config, 0}
 
       :local_over_remote ->
-        with {:ok, remote_settings, fetch_time_ms} <- policy.get(instance_id) do
-          {:ok, Map.merge(remote_settings, local_settings), fetch_time_ms}
+        with {:ok, remote_config, fetch_time_ms} <- policy.get(instance_id) do
+          {:ok, Config.merge(remote_config, local_config), fetch_time_ms}
         end
 
       :remote_over_local ->
-        with {:ok, remote_settings, fetch_time_ms} <- policy.get(instance_id) do
-          {:ok, Map.merge(local_settings, remote_settings), fetch_time_ms}
+        with {:ok, remote_config, fetch_time_ms} <- policy.get(instance_id) do
+          merged = Config.merge(local_config, remote_config)
+          {:ok, merged, fetch_time_ms}
         end
     end
   end
+
+  defp check_type_mismatch(_value, nil), do: :ok
+
+  defp check_type_mismatch(value, default_value) do
+    value_type = SettingType.infer_elixir_type(value)
+    default_type = SettingType.infer_elixir_type(default_value)
+    number_types = ["float()", "integer()"]
+
+    cond do
+      value_type == default_type ->
+        :ok
+
+      value_type in number_types and default_type in number_types ->
+        :ok
+
+      true ->
+        ConfigCatLogger.warning(
+          "The type of a setting does not match the type of the specified default value (#{default_value}). " <>
+            "Setting's type was #{value_type} but the default value's type was #{default_type}. " <>
```

Comment:
Again, the Python SDK uses the Python type in both places. The JS SDK uses the ConfigCat setting type for `value_type`, but the JS type for `default_type`. I'll change Elixir to use the ConfigCat type for both, but let me know which you prefer.

Reply:
Addressed in 4130cee

Reply:
> The JS SDK uses the ConfigCat setting type for value_type, but the JS type for default_type

I think this is the correct approach because the default value is not required to be of a valid setting type.

@kp-cat, would be nice to fix this minor inconsistency in Python SDK as well.

Reply:
Elixir doesn't have a clean way to determine the type of an arbitrary value, so it would be difficult to implement these. There are [workarounds](https://stackoverflow.com/a/40820903), but they're non-standard and potentially fragile.

Is it OK to leave the Elixir implementation as-is?

Reply:
Ok, this is not something critical, so if there's no clean way to do this, let's leave this as-is. 

I talked about this with @kp-cat, and he said that he wouldn't want to change this in Python either.